go-controller/pkg/kubevirt/pod.go

```diff
@@ -503,16 +504,52 @@ func ReconcileIPv4DefaultGatewayAfterLiveMigration(watchFactory *factory.WatchFa
 	}
 
 	lrpMAC := util.IPAddrToHWAddr(lrpJoinIPv4)
-	for _, subnet := range netInfo.Subnets() {
+	for _, subnet := range r.netInfo.Subnets() {
 		gwIP := util.GetNodeGatewayIfAddr(subnet.CIDR).IP.To4()
 		if gwIP == nil {
 			continue
 		}
 		garp := util.GARP{IP: gwIP, MAC: &lrpMAC}
-		if err := util.BroadcastGARP(interfaceName, garp); err != nil {
+		if err := util.BroadcastGARP(r.interfaceName, garp); err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
+
+func (r *DefaultGatewayReconciler) ReconcileIPv6AfterLiveMigration(liveMigration *LiveMigrationStatus) error {
+	if liveMigration.State != LiveMigrationTargetDomainReady {
+		return nil
+	}
+	sourceNode, err := r.watchFactory.GetNode(liveMigration.SourcePod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+	sourceNodeJoinAddrs, err := util.ParseNodeGatewayRouterJoinAddrs(sourceNode, r.netInfo.GetNetworkName())
+	if err != nil {
+		return err
+	}
+	targetNode, err := r.watchFactory.GetNode(liveMigration.TargetPod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+	targetNodeJoinAddrs, err := util.ParseNodeGatewayRouterJoinAddrs(targetNode, r.netInfo.GetNetworkName())
+	if err != nil {
+		return err
+	}
+	return ndp.SendRouterAdvertisments(r.interfaceName,
+		newRouterAdvertisementFromJoinIPAndLifetime(sourceNodeJoinAddrs[0].IP, 0),
+		newRouterAdvertisementFromJoinIPAndLifetime(targetNodeJoinAddrs[0].IP, 65535),
+	)
+}
+
+func newRouterAdvertisementFromJoinIPAndLifetime(ip net.IP, lifetime uint16) ndp.RouterAdvertisment {
```

Comment:
Could you please add a docblock to this function explaining the purpose of using multicast?
I know the function is private but I am sure I will forget the details tomorrow :dizzy: